strapi/strapi
# Average and sum aggregators for Query Engine API

### What does it do?

Adds the `avg` and `sum` functions to the Query Engine API.

### Why is it needed?

Just like the `count` aggregator, it is sometimes useful to retrieve sums and/or averages from the database directly instead of fetching entries and executing operations over them manually.

### How to test it?

1. Create a `Test` content type, with a numeric field called `value`;
2. Create a random amount of entries of type `Test`;
3. Run `strapi.db.query('api::data.data').sum("value");` and notice how it returns the sum of those values;
4. Run `strapi.db.query('api::data.data').avg("value");` and notice how it returns the average of those values.
